packages/typescript/src/internal/ts_project.bzl

```diff
@@ -0,0 +1,225 @@
+"ts_project rule"
+
+load("@build_bazel_rules_nodejs//:providers.bzl", "DeclarationInfo", "NpmPackageInfo", "run_node")
+
+# See https://github.com/microsoft/TypeScript/issues/37378
+# When running in standalone (no sandbox), tsc will see the .ts source files
+# from our dependencies. It then tries to do an up-to-date check on the outputs.
+# Since the outDir can't go in the tsconfig (it depends on platform and config)
+# tsc can't know where to look for the .d.ts outputs and will error that the
+# dependency isn't up-to-date.
+# Error looks like
+#  (05:54:49) ERROR: C:/b/bk-windows-b4qr/bazel/rules-nodejs-nodejs/packages/typescript/test/ts_project/b/BUILD.bazel:6:1: Compiling TypeScript project packages/typescript/test/ts_project/b/tsconfig.json failed (Exit 2): tsc.bat failed: error executing command
+#    cd C:/b/uuxnwop3/execroot/build_bazel_rules_nodejs
+#    SET COMPILATION_MODE=fastbuild
+#    bazel-out/host/bin/external/npm/typescript/bin/tsc.bat -p packages/typescript/test/ts_project/b/tsconfig.json --outDir bazel-out/x64_windows-fastbuild/bin/packages/typescript/test/ts_project/b --declarationDir bazel-out/x64_windows-fastbuild/bin/packages/typescript/test/ts_project/b --bazel_node_modules_manifest=bazel-out/x64_windows-fastbuild/bin/packages/typescript/test/ts_project/b/_tsconfig.module_mappings.json --nobazel_patch_module_resolver
+#  Execution platform: @local_config_platform//:host
+#  packages/typescript/test/ts_project/b/b.ts(1,17): error TS6305: Output file 'C:/b/uuxnwop3/execroot/build_bazel_rules_nodejs/packages/typescript/test/ts_project/a/a.d.ts' has not been built from source file 'C:/b/uuxnwop3/execroot/build_bazel_rules_nodejs/packages/typescript/test/ts_project/a/a.ts'.
+TS_37378_DISABLED = ["fix-windows"]
+
+_ATTRS = {
+    # TODO: limit to the extensions we expect? ts, tsx, js, jsx, json
+    "srcs": attr.label_list(allow_files = True, mandatory = True),
```

Comment:
I don't think we should limit the extensions here, especically if you can swap out the compiler for `ngtsc` for example, since the inputs would then include .html and .css

Reply:
yeah I think even tsc keeps adding ability to natively type-check against require('./funny.thing')